pkg/domain/repository/worker.go

```diff
@@ -187,6 +187,8 @@ func (w *Worker) start(ctx context.Context) func() {
 	return func() {
 		w.owner = w.newOwner(ownerKey, promptKey)
 		ticker := time.NewTicker(time.Second)
+		w.owner.CampaignOwner()
+		defer w.owner.CampaignCancel()
```

Comment:
Should this be in `stop()` instead? Seems like this would prematurely retire the owner while the worker is still active?

Reply:
Yes, house keeper need this. We've assumed that `newOwner` will call `CampaignOwner` by default. And should not call `CampaignOwner` twice.